extensions/ql-vscode/src/query-history.ts

```diff
@@ -318,6 +320,7 @@ export class QueryHistoryManager extends DisposableObject {
     private queryStorageDir: string,
     ctx: ExtensionContext,
     private queryHistoryConfigListener: QueryHistoryConfig,
+    private readonly getCredentials: () => Promise<Credentials>,
```

Comment:
This is the same pattern being used in the constructor to [`DatabasesUI`](https://github.com/github/vscode-codeql/blob/61d4305593a2c001ee3cc5e846ffaee189d1d5ca/extensions/ql-vscode/src/extension.ts#L438).

My original intention was to do this to make testing easier, but since I'm not directly testing this feature maybe I can change it to what you suggest.